### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by Carlos24Rz**
## Navbar

This PR introduces the Navbar to our system. The Navbar content depends on the type of user (Teacher, Student) and it automatically tracks the current url Path to determine which item will be selected. Responsive Navbar was also developed. 

Screenshot 2023-05-01 at 11 32 32

Screenshot 2023-05-01 at 11 32 42

Screenshot 2023-05-01 at 11 33 15

Screenshot 2023-05-01 at 11 33 25

### CODE REVIEW

File: src/components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          </NavLink>
+        </li>
+        <li className={`${styles['link-container']} ${styles.assignments}`}>
+          <NavLink to="/assignment">
+            <IconNotebook />
+            <span>Actividades</span>
+          </NavLink>
+        </li>
+        {children &&
+          Children.map(children, (navLink, index) => (
```
Comment (CarLosVegga) [inline src/components/NavBar/NavBar.tsx:109]: I like this! I didn't know this was possible.
I have a question about this component. For a student, we will call NavBar, and Teacher will call it with props? When I was first reading your code, I thought for a second that both Student and Teacher would have its own NavBar component according to its needs.

File: src/components/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  useEffect(() => {
+    const focusableElements = menuRef.current?.querySelectorAll('a, button');
+    const firstElement =
+      focusableElements && (focusableElements[0] as HTMLElement);
+    const lastElement =
+      focusableElements &&
+      (focusableElements[focusableElements.length - 1] as HTMLElement);
+
+    const focusTrap = (e: KeyboardEvent) => {
```
Comment (CarLosVegga) [inline src/components/NavBar/NavBar.tsx:32]: I will definitely study this so that I can also work on accessibility in our project

**Review by CarLosVegga — APPROVED** [review]
Great work!

File: src/components/NavBar/NavBar.tsx (hunk 1 above)
Comment (Carlos24Rz) [reply src/components/NavBar/NavBar.tsx:109]: The latter is correct. Our root component checks the type of user logged in our system, and based on that we will render a Student or Teacher Navbar. This Navbar is the baseline, so it only includes the options that share both roles. Later we could add more options specifically to each role.

**Pull request merged**